### PR TITLE
Enhance-Phish.AI

### PR DESCRIPTION
**Pull request by orenzohar**
## Status
- [ ] Ready
- [x] In code review
- [ ] In Progress
- [ ] Hold
      Reason for hold:

## Related Issues
fixes: https://github.com/demisto/etc/issues/12418#event-1965832717  

## Description
added generic polling functionality for URLs.  

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
4.0.0

## Must have
- [x] Tests
- [ ] Documentation (with link to it)

## Additional changes
---

### CODE REVIEW

File: Integrations/integration-PhishAI.yml [hunk 1]
```diff
@@ -50,86 +37,141 @@ script:
...
+        }
+        var report = JSON.parse(res.Body);
+        md = tableToMarkdown('Phish.AI Scan report ' + url, report);
+        return {
+            Type: entryTypes.note,
+            Contents: report,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext: {
+                'Url(obj.Data.Url && obj.Data.Url===val.Data.Url)' : {
```
Comment (DeanArbel) [inline Integrations/integration-PhishAI.yml]:
```suggestion
                'URL(val.Data === obj.Data)' : {
```

File: Integrations/integration-PhishAI.yml [hunk 2]
```diff
@@ -50,86 +37,141 @@ script:
...
         return {
-            Type: entryType.error,
-            Contents: JSON.parse(res.Body),
-            HumanReadable: "Scan timed out"
+            Type: entryTypes.note,
+            Contents: report,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext: {
+                'Url(obj.Data.Url && obj.Data.Url===val.Data.Url)' : {
```
Comment (DeanArbel) [inline Integrations/integration-PhishAI.yml]:
```suggestion
                'URL(val.Data === obj.Data)' : {
```

File: Integrations/integration-PhishAI.yml [hunk 3]
```diff
@@ -50,86 +37,141 @@ script:
...
+                    'Data': url
+                },
+                'IP(obj.Hostname && obj.Hostname===val.Hostname)' : {
```
Comment (DeanArbel) [inline Integrations/integration-PhishAI.yml]:
```suggestion
                'IP(val.Hostname === obj.Hostname)' : {
```

File: Integrations/integration-PhishAI.yml [hunk 4]
```diff
@@ -178,4 +222,16 @@ script:
...
+      description: url to check status for
+    outputs:
+    - contextPath: URL.Data
+      description: URL address
+    - contextPath: URL.Status
+    description: checks on url's status, e.g. completed/in progress
+  runonce: false
+releaseNotes: 'Phish.AI enhancement'
+tests:
+  - 'Forgive me for my sins but I did not create any test'
```
Comment (DeanArbel) [inline Integrations/integration-PhishAI.yml]: Weren't there 2 tests?

Comment (orenzohar) [reply Integrations/integration-PhishAI.yml]: fixed

**Pull request merged**